app/src/main/java/com/example/helpq/AdminFragment.java

```diff
@@ -0,0 +1,76 @@
+import android.support.design.widget.BottomNavigationView;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentTransaction;
+import android.view.LayoutInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+
+public class AdminFragment extends Fragment {
+
+    public static final String TAG = "AdminFragment";
+    FragmentManager fragmentManager;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_admin, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        fragmentManager = getFragmentManager();
+        BottomNavigationView navigationView = (BottomNavigationView) view.findViewById(R.id.top_navigation);
+
+        // Set top menu to navigate between "Queue" and "Students" tabs
+        navigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
+            @Override
+            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
+                switch (item.getItemId()) {
+                    case R.id.action_view_queue:
+                        setFragment(new AdminQueueFragment(),
+                                new String[]{AdminQueueFragment.TAG, AdminStudentsFragment.TAG});
+                        return true;
+                    case R.id.action_view_students:
+                        setFragment(new AdminStudentsFragment(),
+                                new String[]{AdminStudentsFragment.TAG, AdminQueueFragment.TAG});
+                        return true;
+                    default:
+                        setFragment(new AdminQueueFragment(),
+                                new String[]{AdminQueueFragment.TAG, AdminStudentsFragment.TAG});
+                        return true;
+                }
+            }
+        });
+        navigationView.setSelectedItemId(R.id.action_view_queue);
+    }
+
+    // Show/add the given fragment with tag stored in tags[0], and hide fragments with all other tags.
+    public void setFragment(Fragment fragment, String[] tags) {
+
+        if (fragmentManager.findFragmentByTag(tags[0]) != null) {
+            // if the fragment exists, show it.
+            fragmentManager.beginTransaction().show(fragmentManager.findFragmentByTag(tags[0])).commit();
+        } else {
+            // if the fragment does not exist, add it to fragment manager.
+            // always add a new fragment if it is a PostDetailsFragment
+            FragmentTransaction transaction = fragmentManager.beginTransaction();
+            transaction.add(R.id.flAdminContainer, fragment, tags[0]).commit();
```

Comment:
I'm sort of confused about what's happening here, but I'm guessing it's because I'm unfamiliar with fragments.

It seems like, we:
1. Try to retrieve and display first fragment.
2. If it's not there, create a fragment and add to manager.
3. Retrieve the other fragments, but hide them.

On step #2, don't we need to call `show` once we've added it to the fragment manager?

Also, it seems like this could be simplified by doing the for loop in the beginning, retrieving all fragments. Then, looping through them to show/hide.